android/src/main/kotlin/com/github/fluttercommunity/flutterdownloader/KotlinDownload.kt

```diff
@@ -0,0 +1,97 @@
+package com.github.fluttercommunity.flutterdownloader
+
+import androidx.annotation.UiThread
+import io.flutter.plugin.common.MethodChannel
+import java.io.File
+import java.net.URL
+
+/**
+ * The Kotlin part of the PlatformDownload from dart. The naming should be quiet similar to the dart and Swift
+ * implementation. Some details will differ since that are different programming languages.
+ */
+internal class KotlinDownload(
```

Comment:
I'd rename this to `AndroidPlatformDownload` (and the iOS/macOS to either `ApplePlatformDownload` or `DarwinPlatformDownload` (or `IOSPlatformDownload` and `MacOSPlatformDOwnload`, respectively)).

Reply:
Good idea so use the Platform in name instead of the Programming language. I'll use something like Dart->Desktop, Swift -> Ios, Kotlin -> Android renamings

Reply:
I still need to think a bit about the names here. Currently I have a lot of "Download" prefixes in class names. I think I'll remove them later.

In the Kotlin code I have now the serializable data class `DownloadMetadata` which is the JSON basically. Then I have also a wrapper `AndroidDownload` which holds the metadata manage the message channel for updates from the platform specific code to the dart code.

Reply:
Sure :)

Here's how I used `Codable` protocol in Swift:

- [Data model definition](https://github.com/leancodepl/patrol/blob/e277beb752138f0175437c95b6ea3fc86132ead7/AutomatorServer/ios/AutomatorServerUITests/PatrolServer.swift#L17-L21)

- [Decoding](https://github.com/leancodepl/patrol/blob/e277beb752138f0175437c95b6ea3fc86132ead7/AutomatorServer/ios/AutomatorServerUITests/PatrolServer.swift#L115)

(This code is not on `master` branch anymore - it was migrated to protobuf for _reasons_)